includes/Core/Assets/Assets.php

```diff
@@ -781,6 +789,39 @@ private function get_inline_dashboard_sharing_data() {
 			}
 		}
 
+		// Expose recoverable modules to the client.
+		$modules           = new Modules( $this->context );
+		$shareable_modules = $modules->get_shareable_modules();
+
+		foreach ( $shareable_modules as $module ) {
+			$owner_id = $module->get_owner_id();
+
+			/**
+			 * If no owner identified by its owner_id OR lacks the AUTHENTICATE capability -
+			 * push the module slug to the recoverableModules array.
+			 */
+			if ( 0 === $owner_id || ! user_can( $owner_id, Permissions::AUTHENTICATE ) ) {
+				$inline_data['recoverableModules'][] = $module->slug;
+				continue;
+			}
+
+			$user = get_user_by( 'ID', $owner_id );
+
+			// If the User doesn't exists - push the module slug to the recoverableModules array.
+			if ( false === $user ) {
+				$inline_data['recoverableModules'][] = $module->slug;
+				continue;
+			}
+
+			$module_owner_user_options   = new User_Options( $this->context, $user->ID );
+			$module_owner_authentication = new Authentication( $this->context, null, $module_owner_user_options );
```

Comment:
You can use `$this->authentication->is_authenticated()` instead of creating new instances once this code is moved to the Modules class. Also you will need to switch the user before checking whether they are authenticated:

```php
// Modules.php
$restore_user = $this->user_options->switch_user( $module_owner_id );
if ( ! $this->authentication->is_authenticated() ) {
    ...
}
$restore_user();
```